usaspending_api/awards/management/commands/load_subawards.py

```diff
@@ -1,185 +1,155 @@
-
-        if self.log_sql:
-            logger.info(sql)
-
-        with Timer("Retrieve incremental values from broker_subaward"):
-            connection = get_connection()
-            with connection.cursor() as cursor:
-                cursor.execute(sql)
-                return cursor.fetchall()[0]
-
-    def _perform_load(self, where):
-        """
-        The actual heavy lifting.  Call the SQLs necessary to load subawards.
-        """
-        if not self.full_reload:
-            self._execute_sql_file("010_find_new_awards.sql")
-
-        with transaction.atomic():
-            if self.full_reload:
-                with Timer("delete from broker_subaward"):
-                    self._execute_sql("delete from broker_subaward")
-            self._execute_sql_file("020_import_broker_subawards.sql", where=where)
-
-        self._execute_sql_file("030_frame_out_subawards.sql")
-        self._execute_sql_file("040_enhance_with_awards_data.sql")
-        self._execute_sql_file("050_enhance_with_transaction_data.sql")
-        self._execute_sql_file("060_enhance_with_cfda_data.sql")
-        self._execute_sql_file("070_enhance_with_awarding_agency.sql")
-        self._execute_sql_file("080_enhance_with_funding_agency.sql")
-        self._execute_sql_file("090_create_temp_address_table.sql")
-        self._execute_sql_file("100_enhance_with_pop_county_city.sql")
-        self._execute_sql_file("110_enhance_with_pop_country.sql")
-        self._execute_sql_file("120_enhance_with_recipient_location_county_city.sql")
-        self._execute_sql_file("130_enhance_with_recipient_location_country.sql")
-
-        # For update_award_totals, if we're performing a full reload, update
-        # all awards that have subawards, otherwise, if this is an incremental
-        # update, just update those awards that have may have been affected.
+        self.full_reload = options["full_reload"]
+
+        with Timer("Load subawards"):
+            try:
+                with transaction.atomic():
+                    self._perform_load()
+                    t = Timer("Committing transaction")
+                    t.log_starting_message()
+                t.log_success_message()
+            except Exception:
+                logger.error("ALL CHANGES WERE ROLLED BACK DUE TO EXCEPTION")
+                raise
+
+    @staticmethod
+    def _execute(message, function, *args, **kwargs):
+        """ Execute an import step. """
+
+        with Timer(message):
+            results = function(*args, **kwargs)
+            if type(results) is int and results > -1:
+                logger.info("{:,} rows affected".format(results))
+            return results
+
+    def _execute_sql_file(self, filename):
+        """ Read in a SQL file and execute it. """
+
+        filepath = (self.sql_path / filename).with_suffix(".sql")
+        return self._execute(filename, execute_update_sql, sql=filepath.read_text())
```

Comment:
Should this still be called `execute_update_sql`? It seems like it's getting more generic

Reply:
`execute_update_sql` is a convenience function intended to be used for running CRUD(-R) SQL commands.  Not sure what a better name would be.   `execute_CUD_sql` maybe?  Yes it can be used for reads, but it shouldn't be since it only acts against the master database instance instead of taking advantage of our read replica(s).